src-ts/Core/CreepSystem/CreepUpgradeConfig.ts

```diff
@@ -0,0 +1,268 @@
+import { RandomCreepType } from "./CreepUpgradeTypes";
+
+export module SagaUpgradeNames {
```

Comment:
not as familiar with module, but if something needs to be of type "SagaUpgradeNames", then you could make this an enum instead. That way the value would have to be 1 of those 4